### PR TITLE
ap-2730: Create document_categories table

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2730)

Created migration to add new document_categories table, model and factory
Created document_category_populator to populate new table

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -103,9 +103,9 @@
     t.string "unlock_token"
     t.datetime "locked_at"
     t.string "true_layer_secure_data_id"
+    t.boolean "employed"
```
Comment (mpw5) [inline db/schema.rb:106]: This change seems to creep in to migrations sometimes. I'm not entirely sure why but it shouldn't really be there. You can undo it by running:

```
rails db:rollback
git checkout origin/main db/schema.rb
rails db:schema:load
rails db:migrate
```
This should reset the schema and then reapply your migration but without the extra changes on lines 106 and 605.

Warning: `rails db:schema:load` will wipe your local db 😬

File: db/schema.rb [hunk 2]
```diff
@@ -592,7 +602,7 @@
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
     t.boolean "scanner_working"
-    t.index ["uploader_type", "uploader_id"], name: "index_malware_scan_results_on_uploader"
+    t.index ["uploader_type", "uploader_id"], name: "index_malware_scan_results_on_uploader_type_and_uploader_id"
```
Comment (mpw5) [inline db/schema.rb:605]: As above.

**Review by mpw5 — COMMENTED** [review]
Looks good but some unrelated schema changes have crept in. Happy to approve once they've been fixed. I've added some comments to help fix it but give me a shout if you want a hand.

**Review by mpw5 — APPROVED** [review]
LGTM!

File: db/schema.rb (hunk 2 above)
Comment (kmahern) [reply db/schema.rb]: Thanks @mpw5, I also don't understand how the unrelated schema changes crept in (I definitely pulled the version from main before I started!) but your suggestion removed them anyway.

**Pull request merged**